repose-aggregator/components/services/http-logging-service/http-logging-service-impl/src/main/scala/org/openrepose/core/services/httplogging/HttpLoggingConfigListener.scala

```diff
@@ -0,0 +1,115 @@
+/*
+ * _=_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=
+ * Repose
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Copyright (C) 2010 - 2015 Rackspace US, Inc.
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package org.openrepose.core.services.httplogging
+
+import com.fasterxml.jackson.core.JsonProcessingException
+import com.typesafe.scalalogging.slf4j.StrictLogging
+import javax.inject.Named
+import org.jtwig.environment.{EnvironmentConfiguration, EnvironmentConfigurationBuilder}
+import org.jtwig.exceptions.JtwigException
+import org.jtwig.value.Undefined
+import org.jtwig.value.convert.string.{DefaultStringConverter, StringConverter}
+import org.jtwig.{JtwigModel, JtwigTemplate}
+import org.openrepose.commons.config.manager.UpdateListener
+import org.openrepose.core.services.httplogging.config.{Format, HttpLoggingConfig, Message}
+import org.openrepose.core.services.httplogging.jtwig.HttpLoggingEnvironmentConfiguration
+import org.slf4j.LoggerFactory
+import play.api.libs.json.Json
+
+import scala.collection.JavaConverters._
+import scala.util.{Success, Try}
+
+/**
+  * Handles configuration for the [[HttpLoggingService]].
+  */
+@Named
+class HttpLoggingConfigListener extends UpdateListener[HttpLoggingConfig] with StrictLogging {
```

Comment:
Indeed. I messed around with the organization here for a while, but ultimately I think this is what works best. Not only because of the size of the class, but also (and relatedly) because I can use dependency injection to unit test both this and the service.